packages/toolpad-app/src/toolpad/AppEditor/PagesExplorer/index.tsx

```diff
@@ -224,41 +337,49 @@ export default function PagesExplorer({ className }: PagesExplorerProps) {
       className={className}
     >
       <TreeView
-        aria-label="pages explorer"
+        ref={pagesTreeRef}
+        aria-label="Pages explorer"
         selected={activeNode ? [activeNode] : []}
         onNodeSelect={handleSelect}
         expanded={expanded}
         onNodeToggle={handleToggle}
         multiSelect
         defaultCollapseIcon={<ExpandMoreIcon sx={{ fontSize: '0.9rem', opacity: 0.5 }} />}
         defaultExpandIcon={<ChevronRightIcon sx={{ fontSize: '0.9rem', opacity: 0.5 }} />}
+        sx={{
+          scrollbarGutter: 'stable',
+        }}
       >
-        <PagesExplorerTreeItem
-          nodeId=":pages"
-          aria-level={1}
-          labelText="Pages"
-          createLabelText="Create page"
-          onCreate={handleCreatePageDialogOpen}
-        >
-          {pages.map((page) => (
-            <PagesExplorerTreeItem
-              key={page.id}
-              nodeId={page.id}
-              toolpadNodeId={page.id}
-              aria-level={2}
-              labelText={page.name}
-              onDuplicateNode={handleDuplicateNode}
-              onDeleteNode={handleDeletePage}
-            />
-          ))}
-        </PagesExplorerTreeItem>
+        <ExplorerHeader
+          headerText="Pages"
+          onCreate={handleOpenCreateNewPage}
+          createLabelText="Create new page"
+        />
+        {isCreateNewPageOpen ? (
+          <EditableTreeItem
+            nodeId="::create::"
+            isEditing
+            suggestedNewItemName={nextProposedName}
+            onEdit={handleCreateNewCommit}
+            onCancel={handleCloseCreateNewPage}
+            validateItemName={validatePageName}
+            sx={pagesTreeItemSx}
+          />
+        ) : null}
+        {sortBy(pages, 'name').map((page) => (
```

Comment:
This is mutating the `pages` array. Best to Avoid mutating variables in the render function.
It's also running a sort i every render. Feels a bit wasteful.

Reply:
Yeah, I noticed it was mutable, in this particular case it doesn't have too much impact but it's better to make it immutable and memoize it, I'll do it.